### ISSUE TITLE
Loss scaling in ComputeLoss.

### ISSUE THREAD

**Issue by Itto1992** [ultralytics/yolov5#2704]
## ❔Question
Thank you for your creating such a nice repo!

I want to know why the loss is scaled by batch size at the last line of `ComputeLoss`.
https://github.com/ultralytics/yolov5/blob/9ccfa85249a2409d311bdf2e817f99377e135091/utils/loss.py#L161

In my understanding, the loss value has already been summed up across a batch and proportional to the batch size. Is is a special technique for training yolov5?

Thanks!

**Comment by glenn-jocher** [comment]
@Itto1992 yes the loss is scaled by batch size before optimizing! The loss is scaled by batch size to allow everyone to achieve the same results no matter what batch they use. This way people on smaller GPUs (that can only fit smaller batches) should be able to achieve comparable results to people with expensive larger GPUs that can fit larger batches.

This also means you can use most any batch size without having to worry about it's effect on your training. We actually performed a recent study verifying this is practice:

**🌟💡 YOLOv5 Study**: batch size #2452

**Comment by Itto1992** [comment]
@glenn-jocher 

Thanks for your reply!
I interpreted this implementation as the linear rule on learning rate and batch size stated in [this paper](https://arxiv.org/pdf/1711.00489.pdf), since loss scaling has the same effect as scaling learning rate. This paper says the learning rate should be proportional to the batch size. From this point of view, the loss scale (= learning rate) seems to be proportional to the square of the batch size. I suspected that it can result in too large gradient and unstable learning behavior. But your study shows it is needless fear! 

I'm looking forward to reading the paper of this repo to be published :)
Thanks!
